nixos/modules/services/web-servers/webhook.nix

```
@@ -0,0 +1,91 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.webhook;
+  webhook = pkgs.webhook;
+  user = if (cfg.user == null) then "webhook" else cfg.user;
+  group = if (cfg.group == null) then "webhook" else cfg.group;
+  settingsFormat = pkgs.formats.json {};
+  configFile = settingsFormat.generate "webhook.json" cfg.settings;
+
+in {
+  options.services.webhook = {
+
+    enable = mkEnableOption "the webhook service";
+
+    user = mkOption {
+      default = null;
+      description = "User webhook service runs as. If <literal>null</literal>, it will default to <literal>webhook</literal>";
+      type = with types; nullOr str;
+    };
+
+    group = mkOption {
+      default = null;
+      description = "Group webhook service runs as. If <literal>null</literal>, it will default to <literal>webhook</literal>";
+      type = with types; nullOr str;
+    };
+
+    settings = mkOption {
+      description = "Webhook service settings.";
+      type = settingsFormat.type;
+      example = literalExpression ''
+        [{
+          id = "redeploy-webhook";
+          execute-command = "/var/scripts/redeploy.sh";
+          command-working-directory = "/var/webhook";
+        }]
+      '';
+    };
+
+    path = mkOption {
+      default = [];
+      description = "Which packages the service can access";
+      example = literalExpression ''
+        with pkgs; [ nixFlakes ]
+      '';
+      type = with types; listOf package;
+    };
+
+    extraOptions = mkOption {
+      default = [ "-verbose" ];
+      description = "Which flags to append to the command";
+      example = literalExpression ''
+        [ "-debug" "-port 12345" ]
```

```suggestion
        [ "-debug" "-port" "12345" ]
```
it's usually better to have one argument per element, then escape the command line to behave accordingly. otherwise surprising things can happen.

---

My concern was just that if lists are merged, they might get reordered.

---

each individual list will retain its order, but different lists may be concatenated in any order. if order is important there's `mkOrder`/`mkBefore`/`mkAfter` to ensure proper ordering